### PR TITLE
Fix some typos

### PR DESCRIPTION
**Pull request by nlamprian**

### CODE REVIEW

File: plugin_tutorials/docs/writing_new_behavior_plugin.rst [hunk 1]
```diff
@@ -42,27 +42,27 @@ This wrapper class derives from the ``nav2_core`` class so it can be used as a p
...
 The plugin will be used by the behavior server to host the plugins, but each plugin will provide their own unique action server interface.
-Lets learn more about the methods needed to write a Behavior Plugin **if you did not use the ``nav2_behaviors`` wrapper**.
+Let's learn more about the methods needed to write a Behavior Plugin **if you did not use the ``nav2_behaviors`` wrapper**.
 
 +----------------------+----------------------------------------------------------------------------+-------------------------+
 | **Virtual method**   | **Method description**                                                     | **Requires override?**  |
 +----------------------+----------------------------------------------------------------------------+-------------------------+
 | configure()          | Method is called at when server enters on_configure state. Ideally         | Yes                     |
-|                      | this methods should perform declarations of ROS parameters and             |                         |
+|                      | this method should perform declarations of ROS parameters and             |                         |
```
Comment (SteveMacenski) [inline plugin_tutorials/docs/writing_new_behavior_plugin.rst:51]: For here and all tables below, the table lines must align in restructured text. So these word changes require properly formatting the table

File: plugin_tutorials/docs/writing_new_behavior_plugin.rst [hunk 2]
```diff
@@ -81,17 +81,17 @@ This tutorial uses this wrapper so these are the main elements we will address.
...
+|                      | behavior goes to a configured state (get parameters, etc).                |                         |
 +----------------------+----------------------------------------------------------------------------+-------------------------+
 | onCleanup()          | Method is called when behavior server goes to on_cleanup state. Ideally    | No                      |
-|                      | this method should clean up resoures which are created for the behavior.   |                         |
+|                      | this method should clean up resources which are created for the behavior.   |                         |
 +----------------------+----------------------------------------------------------------------------+-------------------------+
 
 For this tutorial, we will be using methods ``onRun()``, ``onCycleUpdate()``, and ``onConfigure()`` to create the SMS behavior.
-``onConfigure()`` will be skipped for brevity, but only includes declaring of parameters.
+``onConfigure()`` will be skipped for brevity, but only declares of parameters.
```
Comment (SteveMacenski) [inline plugin_tutorials/docs/writing_new_behavior_plugin.rst:91]:
```suggestion
``onConfigure()`` will be skipped for brevity, but only declares parameters.
```

File: tutorials/docs/using_groot.rst [hunk 3]
```diff
@@ -27,12 +27,12 @@ Overview
...
 Behavior Trees are deeply integrated into Nav2, used as the main method of orchestrating task server logic across a complex navigation and autonomy stack.
-Behavior Trees, in short BTs, consist of many nodes completing different tasks and control the flow of logic, similar to a Heirarchical or Finite State Machine, but organized in a tree structure.
-These nodes are of types: `Action`, `Condition`, `Control`, or `Decorator`, and are described in more detail in :ref:`concepts` and `BehaviorTree.CPP <https://github.com/BehaviorTree/BehaviorTree.CPP>`_.
+Behavior Trees, in short BTs, consist of many nodes completing different tasks and control the flow of logic, similar to a Hierarchical or Finite State Machine, but organized in a tree structure.
+These nodes are of types: `Action`, `Condition`, `Control`, or `Decorator`, and are described in more detail in `Introduction To Nav2 Specific Nodes <https://navigation.ros.org/behavior_trees/overview/nav2_specific_nodes.html>`_ and `BehaviorTree.CPP <https://www.behaviortree.dev/docs/learn-the-basics/BT_basics#types-of-nodes>`_.
```
Comment (SteveMacenski) [inline tutorials/docs/using_groot.rst:31]: Revert back to use of ref to concepts page, that should always be used for internal pages. The BT change makes sense

**Review by SteveMacenski — COMMENTED** [review]
Thanks so much for that diligent work!

File: plugin_tutorials/docs/writing_new_nav2controller_plugin.rst [hunk 4]
```diff
@@ -65,25 +65,25 @@ The list of methods, and their descriptions, and necessity are presented in the
...
 | setPlan()                 | Method is called when the global plan is updated. Ideally this method should perform  | Yes                    |
-|                           | operations that transform the global plan and stores it.                              |                        |
+|                           | operations that transform the global plan and store it.                               |                        |
 +---------------------------+---------------------------------------------------------------------------------------+------------------------+
 | computeVelocityCommands() | Method is called when a new velocity command is demanded by the controller server     | Yes                    |
 |                           | in-order for the robot to follow the global path. This method returns a               |                        |
-|                           | `geometry_msgs::msg::TwistStamped` which represents the velocity command for the      |                        |
+|                           | `geometry_msgs\:\:msg\:\:TwistStamped` which represents the velocity command for the  |                        |
```
Comment (nlamprian) [inline plugin_tutorials/docs/writing_new_nav2controller_plugin.rst:72]: I didn't test this. Highlighting appeared broken in my editor, so I escaped the colons.

**Comment by nlamprian** [comment]
Checked the latest build; good to go.

The escapes didn't fix the rendering, but at least they didn't break anything. Looks like the solution is more involved, https://stackoverflow.com/a/66490390

**Comment by SteveMacenski** [comment]
Thanks so much!

I tend to write alot of these docs without a proofing run because I figure 3x the documentation is better than 1/3rd as much but proof read to perfection.

**Comment by nlamprian** [comment]
No worries. I went through almost everything, and I can safely say every little bit counts. The framework is extensible, so the doc is much appreciated.

**Pull request merged**